### PR TITLE
Modify all calls to `WriterWriteHelp` to adapt to API change

### PR DESCRIPTION
**Pull request by lluiscampos**
Merge together:
https://github.com/cfengine/core/pull/5339
https://github.com/cfengine/nova/pull/2128

### CODE REVIEW

**Comment by lluiscampos** [comment]
Depends on:
* https://github.com/cfengine/libntech/pull/194

File: cf-agent/cf-agent.c [hunk 1]
```diff
@@ -179,6 +179,13 @@ static const char *const CF_AGENT_MANPAGE_LONG_DESCRIPTION =
         "For each bundle, cf-agent groups promise statements according to their type. Promise types are then evaluated in a preset "
         "order to ensure fast system convergence to policy.\n";
 
+static const Component COMPONENT =
+{
+    .name = "cf-agent",
+    .website = "https://cfengine.com",
+    .copyright = "2023 Northern.tech AS"
```
Comment (vpodzime) [inline cf-agent/cf-agent.c:186]: Website and copyright should be macros defined in either `cf3.defs.h` or in `config.h` (by means of autotools). We don't want to update them in multiple places.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise. It will need changes in other CFEngine repos, but I can take care of that. Thanks!

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9937/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9937/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9937/

**Review by olehermanse — CHANGES_REQUESTED** [review]
Great work, but looks like this needs at least a change in enterprise repo for windows to compile.

**Comment by vpodzime** [comment]
> Great work, but looks like this needs at least a change in enterprise repo for windows to compile.

What do you mean @olehermanse? I see this:
>  ci/testing_pr/PACKAGES_i386_mingw — Build and tests finished: success 
>  ci/testing_pr/PACKAGES_x86_64_mingw — Build and tests finished: success

**Comment by larsewi** [comment]
Tested, built together with https://github.com/cfengine/nova/pull/2130#issuecomment-1754585718. And it seems to work :)

**Review by larsewi — APPROVED** [review]

**Comment by larsewi** [comment]
> Merging can be performed automatically once the requested changes are addressed.

Not able to merge

**Review by olehermanse — APPROVED** [review]

**Comment by craigcomstock** [comment]
@lluiscampos @olehermanse did we really want to remove all the Commands: text from the help?

It used to be
```
$ /var/cfengine/bin/cf-check --help

cf-check:
        Utility for diagnosis and repair of local CFEngine databases.
        This BETA version of the tool is for testing purposes only.

Commands:
        dump - Print the contents of a database file
        diagnose - Assess the health of one or more database files
        backup - Copy database files to a timestamped folder
        repair - Diagnose, then backup and delete any corrupt databases
        version - Print version information
        help - Print this help menu

Usage:
        $ cf-check <command> [options] [file ...]

Examples:
        $ cf-check dump /var/cfengine/state/cf_lastseen.lmdb
        $ cf-check lmdump -a /var/cfengine/state/cf_lastseen.lmdb
        $ cf-check diagnose
        $ cf-check repair
```

But now is
```
$ cf-check --help
Usage: cf-check [OPTIONS] [FILE]

Options:
  --help        , -h value - Print the help message
  --manpage     , -M       - Print the man page
  --version     , -V       - Output the version of the software
  --debug       , -d       - Enable debugging output
  --verbose     , -v       - Enable verbose output
  --log-level   , -g value - Specify how detailed logs should be. Possible values: 'error', 'warning', 'notice', 'info', 'verbose', 'debug'
  --inform      , -I       - Enable basic information output

Website: https://cfengine.com
This software is Copyright 2023 Northern.tech AS.
```

Not quite as informative as it should be I think.

**Comment by lluiscampos** [comment]
> @lluiscampos @olehermanse did we really want to remove all the Commands: text from the help?
> 

Thanks for reporting.

The commands were indeed removed by mistake. See the follow-up in #5353. See discussion there for what happened to the examples.

**Pull request merged**